### ISSUE TITLE
twisted-dev.el should provide a custom variable to specify its tbformat

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2666]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2666|
|Type|enhancement|
|Created|2007-05-20 17:39:54Z|
|Branch|https://github.com/twisted/twisted/tree/customize-tbformat-2666|

The default format is better, as it includes source lines.

<details><summary>Searchable metadata</summary>

```
trac-id__2666 2666
type__enhancement enhancement
reporter__exarkun exarkun
priority__low low
milestone__ 
branch__branches_customize_tbformat_2666 branches/customize-tbformat-2666
branch_author__radix__thijs radix, thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1179682794000000 1179682794000000
changetime__1295708960000000 1295708960000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

The default format is worse, since it includes redundant information and makes tracebacks twice as long.  You're sitting in emacs, if you want to see the source lines you can click on them or run next-error.

I'll concede that if we have multiple traceback formats that emacs supports, the user should be able to switch between them.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @radix|
|-|-|

Your objections are valid.

I am not going to implement a custom variable, though.

I guess radix will have a fun time implementing this feature.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_3"></a>|@radix set owner to @exarkun|
|-|-|

ready for review in customize-tbformat-2666


what's the coding standard for elisp?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @radix|
|-|-|

Coding standard for elisp is stop changing it until it has unit tests.

Branch looks fine.  Tested customizing the variable and looking at tracebacks, everything seems to work.  Go ahead and merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Can somebody merge this! :D

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [30559]) Add news file, refs [#2666](https://github.com/twisted/twisted/issues/2666)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_4):
> Branch looks fine.  Tested customizing the variable and looking at tracebacks, everything seems to work.  Go ahead and merge.

I guess the news file is only up for review..

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

The contents of misc files are entirely ignored.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [30560]) Merge customize-tbformat-2666: Added ability to specify the tbformat in twisted-dev.el using a custom variable.

Author: radix
Reviewer: exarkun, thijs
Fixes: [#2666](https://github.com/twisted/twisted/issues/2666)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
